examples/openvino/executor_runner/openvino_executor_runner.cpp

```diff
@@ -133,135 +234,53 @@ int main(int argc, char** argv) {
+
+  double elapsed_time;
+  Error status = Error::Ok;
 
   // If the input path list is provided, read input tensors from the files
-  if (!(FLAGS_input_list_path.empty())) {
-    const char* input_list_path = FLAGS_input_list_path.c_str();
-    ET_LOG(Info, "Loading input tensors from the list provided in %s.", input_list_path);
-    Error status = Error::Ok;
-    std::vector<EValue> inputs(method->inputs_size());
-    ET_LOG(Info, "%zu inputs: ", inputs.size());
-    status = method->get_inputs(inputs.data(), inputs.size());
-    ET_CHECK(status == Error::Ok);
-
-    auto split = [](std::string s, std::string delimiter) {
-      size_t pos_start = 0, pos_end, delim_len = delimiter.length();
-      std::string token;
-      std::vector<std::string> res;
-
-      while ((pos_end = s.find(delimiter, pos_start)) != std::string::npos) {
-        token = s.substr(pos_start, pos_end - pos_start);
-        pos_start = pos_end + delim_len;
-        res.push_back(token);
-      }
-      res.push_back(s.substr(pos_start));
-      return res;
-    };
-
-    // Read raw input tensor file names from input list file and
-    // iterate each raw input tensor file to read values
-    std::ifstream input_list(input_list_path);
-    if (input_list.is_open()) {
-      size_t num_inputs = method->inputs_size();
-      std::string file_path;
-      while (std::getline(input_list, file_path)) {
-        auto input_files = split(file_path, " ");
-        if (input_files.size() == 0) {
-          break;
-        }
-        for (int input_index = 0; input_index < num_inputs; ++input_index) {
-            MethodMeta method_meta = method->method_meta();
-            Result<TensorInfo> tensor_meta =
-                method_meta.input_tensor_meta(input_index);
-            auto input_data_ptr = inputs[input_index].toTensor().data_ptr<char>();
-
-            std::ifstream fin(input_files[input_index], std::ios::binary);
-            fin.seekg(0, fin.end);
-            size_t file_size = fin.tellg();
-
-            ET_CHECK_MSG(
-                file_size == tensor_meta->nbytes(),
-                "Input(%d) size mismatch. file bytes: %zu, tensor bytes: %zu",
-                input_index,
-                file_size,
-                tensor_meta->nbytes());
-
-            fin.seekg(0, fin.beg);
-            fin.read(
-                static_cast<char*>(input_data_ptr),
-                file_size);
-            fin.close();
-        }
-      }
-    } else {
-      ET_CHECK_MSG(false,
-          "Failed to read input list file: %s",
-          input_list_path);
+  if (!(FLAGS_input_list_path.empty()) and
+      !(FLAGS_output_folder_path.empty())) {
+    const char *input_list_path = FLAGS_input_list_path.c_str();
+    ET_LOG(Info, "Loading input tensors from the list provided in %s.",
+           input_list_path);
+    const char *output_folder_path = FLAGS_output_folder_path.c_str();
+    auto res = process_inputs(method, input_list_path, output_folder_path);
```

Comment:
Why there is a need to have a separate flows of full execution when input list is provided and when it is not provided? Can't we have a call for input processing for reading inputs only, then a common benchmark_method call out of if-else statements?

Also, in this case, it seems to save the output tensors if input file list is provided but output folder is not provided. But if input file list is not provided, it will save the output tensors only if output folder is provided. I think this could be confusing for the user.

Reply:
Fixed, please check

Reply:
Looks great, thanks!